pkg/utils/helper.go

```diff
@@ -102,7 +102,12 @@ func (s *SecurityGroupForPods) GetMatchingSecurityGroupForPods(pod *corev1.Pod)
 		return nil, err
 	}
 
-	sgList := s.filterPodSecurityGroups(sgpList, pod, sa)
+	sgList, err := s.filterPodSecurityGroups(sgpList, pod, sa)
+	if err != nil {
+		helperLog.Error(err, "Failed in associating pod to security groups")
```

Comment:
nit: error can be more accurate like `"failed to get security groups to be associated with pod", "pod", pod.Name`, since we are not yet creating the branch ENI here.

Reply:
Updated with suggestion